src/components/Card.js

```diff
@@ -8,27 +8,60 @@
 // import PopupWithImage from "../scripts/PopupWithImage.js";
 
 export default class Card {
-  constructor({ name, link }, cardSelector, handleImageClick) {
-    this._name = name;
-    this._link = link;
+  constructor(
+    cards,
+    cardSelector,
+    handleImageClick,
+    handleCardDeleteClick,
+    handleCardLike,
+    _handleCardLikeRemove
+  ) {
+    this._name = cards.name;
+    this._link = cards.link;
+    this._id = cards._id;
+    this._isLiked = cards.isLiked;
     this._cardSelector = cardSelector;
     this._handleImageClick = handleImageClick;
+    this._handleCardDeleteClick = handleCardDeleteClick;
+    this._handleCardLike = handleCardLike;
+    this._handleCardLikeRemove = _handleCardLikeRemove;
+  }
+  getId() {
+    return this._id;
   }
   _setEventListeners() {
     // ---- card like button -----------------------------------------------------------------
 
     this._cardElement
       .querySelector(".card-heart")
       .addEventListener("click", () => {
-        this._handleLikeIcon();
+        if (this._isLiked) {
+          this._handleCardLikeRemove(this);
+          this._isLiked = !this._isLiked;
+        } else {
+          this._handleCardLike(this);
+        }
       });
 
     // ---- card delete button ---------------------------------------------------------------
 
     this._cardElement
       .querySelector(".card__delete-button")
       .addEventListener("click", () => {
-        this._handleDeleteCard();
+        // this._deleteConfirmation.classList.add("modal_opened");
+        // const isConfirmed = document.querySelector("#confirmDeletion");
+        // if (isConfirmed) {
+        //   console.log("Deletion confirmed!");
+        //   this._handleDeleteCard();
+        //   this._handleCardDeleteClick({ name: this._name, link: this._link });
+        //   this._deleteConfirmation.classList.remove("modal_opened");
+        // } else {
+        //   console.log("Deletion canceled.");
+        //   this._deleteConfirmation.classList.remove("modal_opened");
+        // }
+
+        // this._handleDeleteCard();
+        this._handleCardDeleteClick(this);
       });
     this._cardElement
       .querySelector(".card-image")
```

Comment:
You've found `this._cardImageEl` in line 116. Please, use it here


`Class fields` (the so called **class variables** with `this`) are accessible inside any method of the class. You shouldn’t pass **class fields** into the call of any method or search them again because you can use class fields inside all methods.